lib/floki/html_tree.ex

```diff
@@ -99,14 +99,18 @@ defmodule Floki.HTMLTree do
   defp build_tree(tree, [], _, []), do: tree
   defp build_tree(tree, [{tag, attrs, child_children} | children], parent_id, stack) do
     new_id = IDSeeder.seed(tree.node_ids)
-    new_node = %HTMLNode{type: tag,
-                         attributes: attrs,
-                         node_id: new_id,
-                         parent_node_id: parent_id}
+    nodes = if tag != :pi do
```

Comment:
I implemented, tested, works and is definitely a more idiomatic way to solve it.